rfc/20250317-ephemeral-resources.md

```diff
@@ -0,0 +1,646 @@
+# Ephemeral resources, variables, outputs, locals and write-only arguments
+
+Issue: https://github.com/opentofu/opentofu/issues/1996
+
+Right now, OpenTofu information for resources and outputs are written to state as it is. This is presenting a security risk
+as some of the information from the stored objects can contain sensitive bits that can become visible to whoever is having access to the state file.
+
+In order to provide a better solution for the aforementioned situation, OpenTofu introduces the concept of "ephemerality".
+Any new feature under this new concept will provide ways to skip values from being written to the state and plan files.
+
+To make this work seamlessly with most of the blocks that OpenTofu supports, the following functionalities need to be able to work with the ephemeral concept:
+* `resource`'s `write-only` attributes
+* variables
+* outputs
+* locals
+* `ephemeral` resources
```

Comment:
I just want to go on record that I don't like how they're called `ephemeral resources` because this is confusing. Mainly because `resource` and `ephemeral` here are 2 different types of blocks.

I would prefer a different name, such as "ephemeral entities" etc, but I fear it may be too late.

Reply:
I don't think that it's too late and I agree that it can become confusing. If others agree with this, it's an easy change:
* "%s/ephemeral resource/ephemeral entity/g"
* "%s/ephemeral resources/ephemeral entities/g"

Hitting enter? 😅